common/helpers/g2p.py

```diff
@@ -0,0 +1,102 @@
+__all__ = ["G2PBasedOovAugmenter"]
+
+import os
+from typing import Optional
+
+from sisyphus import tk
+Path = tk.setup_path(__package__)
+
+from i6_core.corpus.stats import ExtractOovWordsFromCorpusJob
+from i6_core.g2p.apply import ApplyG2PModelJob
+from i6_core.g2p.convert import BlissLexiconToG2PLexiconJob, G2POutputToBlissLexiconJob
+from i6_core.g2p.train import TrainG2PModelJob
+
+
+class G2PBasedOovAugmenter():
+    def __init__(self,
+                 original_bliss_lexicon,
+                 g2p_model_path:Optional[str] = None,
+                 train_args:Optional[dict] = None,
+                 apply_args:Optional[dict] = None,
+                 ):
+        """
+        :param original_bliss_lexicon: path to the original lexicon with OOV
+        :param g2p_model_path: path to the g2p model, if none a g2p model is trained
+        #######################################
+        :param train_args = {
+        "num_ramp_ups"   :4,
+        "min_iter"       :1,
+        "max_iter"       :60,
+        "devel"          :"5%",
+        "size_constrains":"0,1,0,1"
+        }
+        #######################################
+        :param apply_args = {
+        "variants_mass"  :1.0,
+        "variants_number":1
+        }
+        """
+        super().__init__()
+        self.original_bliss_lexicon = original_bliss_lexicon
+        self.g2p_model_path = g2p_model_path
+        self.train_args = {}
+        self.apply_args = {}
+
+        if train_args is not None:
+            self.train_args.update(train_args)
+
+        if apply_args is not None:
+            self.apply_args.update(apply_args)
+
+
+    def train_and_set_g2p_model(self, train_lexicon:str, alias_path:str):
```

Comment:
you could make this private and call it in the init if `g2p_model_path` is not set.